dnf-behave-tests/features/install-xml-base.feature

```diff
@@ -0,0 +1,85 @@
+  And I copy repository "dnf-ci-fedora" for modification
+  And I execute "createrepo_c --baseurl http://localhost:{context.dnf.ports[dnf-ci-fedora]} ." in "/{context.dnf.repos[dnf-ci-fedora].path}"
+  And I use repository "dnf-ci-fedora"
+ When I execute dnf with args "--setopt=keepcache=true install setup"
+ Then the exit code is 0
+  And stderr is empty
+  And Transaction is following
+      | Action        | Package                                  |
+      | install       | setup-0:2.12.1-1.fc29.noarch             |
+  And glob "/var/cache/dnf/dnf-ci-fedora*/packages/setup*" matches existing file
+
+
+@fixture.httpd
+Scenario: Install from remote repodata with xml:base pointing to different remote packages
+Given I make packages from "dnf-ci-fedora" repository accessible via http
+  And I copy repository "dnf-ci-fedora" for modification
+  And I execute "createrepo_c --baseurl http://localhost:{context.dnf.ports[dnf-ci-fedora]} ." in "/{context.dnf.repos[dnf-ci-fedora].path}"
+  And I use repository "dnf-ci-fedora" as http
+ When I execute dnf with args "install setup"
+ Then the exit code is 0
+  And stderr is empty
+  And Transaction is following
+      | Action        | Package                                  |
+      | install       | setup-0:2.12.1-1.fc29.noarch             |
+
+
+@not.with_os=rhel__eq__8
+@fixture.httpd
+Scenario: Install from local repodata with xml:base pointing to remote packages doesn't delete unused local packages
+Given I make packages from "dnf-ci-fedora" repository accessible via http
+  And I copy repository "dnf-ci-fedora" for modification
+  And I execute "createrepo_c --baseurl http://localhost:{context.dnf.ports[dnf-ci-fedora]} ." in "/{context.dnf.repos[dnf-ci-fedora].path}"
+  And I use repository "dnf-ci-fedora"
+  And file "/{context.dnf.repos[dnf-ci-fedora].path}/noarch/setup-2.12.1-1.fc29.noarch.rpm" exists
+ When I execute dnf with args "install setup"
+ Then file "/{context.dnf.repos[dnf-ci-fedora].path}/noarch/setup-2.12.1-1.fc29.noarch.rpm" exists
+  And the exit code is 0
+  And stderr is empty
+  And Transaction is following
+      | Action        | Package                                  |
+      | install       | setup-0:2.12.1-1.fc29.noarch             |
+
+ 
+@not.with_os=rhel__eq__8
+@fixture.httpd
+Scenario: Install from local repodata that have packages with xml:base pointing to a remote as well as local packages
+Given I make packages from "dnf-ci-fedora" repository accessible via http
+  And I copy repository "dnf-ci-fedora" for modification
+  And I copy repository "dnf-ci-thirdparty" for modification
+  And I execute "createrepo_c --baseurl http://localhost:{context.dnf.ports[dnf-ci-fedora]} ." in "/{context.dnf.repos[dnf-ci-fedora].path}"
+  And I execute "mergerepo_c --repo file://{context.dnf.repos[dnf-ci-fedora].path} --repo file://{context.dnf.repos[dnf-ci-thirdparty].path}" in "{context.dnf.installroot}"
+  And I use repository "dnf-ci-fedora" with configuration
```

Comment:
This is sort-of abusing the "I use repository ..." step by using a different repo and injecting your baseurl... Seeing this I've realized there are other places where this sort of thing is required and whipped up https://github.com/rpm-software-management/ci-dnf-stack/pull/652, which adds a step to create a custom repo config.